sdcm/cluster.py

```diff
@@ -650,21 +689,19 @@ def _notify_backtrace(self, last):
         result = self._get_coredump_backtraces(last=last)
         log_file = os.path.join(self.logdir, 'coredump.log')
         output = result.stdout + result.stderr
-        base_upload_url = 'scylladb-users-upload.s3.amazonaws.com/%s/%s'
         for line in output.splitlines():
             line = line.strip()
             if line.startswith('Coredump:'):
                 coredump = line.split()[-1]
-                coredump_id = os.path.basename(coredump)[:-3]
-                upload_url = base_upload_url % (coredump_id,
-                                                os.path.basename(coredump))
-                self.log.info('Uploading coredump %s to %s' %
-                              (coredump, upload_url))
-                self.remoter.run("sudo curl --request PUT --upload-file "
-                                 "'%s' '%s'" % (coredump, upload_url))
-                self.log.info("To download it, you may use "
-                              "'curl --user-agent [user-agent] %s > %s'",
-                              upload_url, os.path.basename(coredump))
+                self.log.debug('Found coredump file: {}'.format(coredump))
+                file_size = self._get_coredump_size(coredump)
+                if file_size and file_size > COREDUMP_MAX_SIZE:
+                    coredump_files = self._split_coredump(coredump)
+                else:
+                    coredump_files = [coredump]
+                for f in coredump_files:
+                    self._upload_coredump(f)
```

Comment:
thank you for your comment :), but I personally prefer python code cause it's more consistent and easier to debug. I would write the shell script only if it's more then 3-4 commands or for better performance in case ssh communication is very slow.